### ISSUE TITLE
pointer operator with function

### ISSUE THREAD

**Issue by Konstantin8105** [elliotchance/c2go#494]
C:
```c
#include <string.h>

#undef strcat

#ifndef STRCAT
# define STRCAT strcat
#endif

/* Append SRC on the end of DEST.  */
char *
STRCAT (char *dest, const char *src)
{
  strcpy (dest + strlen (dest), src);
  return dest;
}
```
Go:
```go
package main

import "github.com/elliotchance/c2go/noarch"

type size_t uint32
type __locale_data struct {
}
type __locale_t struct {
        __locales       [][]__locale_data
        __ctype_b       []uint16
        __ctype_tolower []int
        __ctype_toupper []int
        __names         [][]byte
}
type locale_t __locale_t

func strcat(dest []byte, src []byte) []byte {
        strcpy(dest+noarch.IntToByteSlice(noarch.Strlen(dest)), src) // <--- WRONG HERE
        return dest
}
func __init() {
}
```

**Comment by Konstantin8105** [comment]
```c
char *
STRNCAT (char *s1, const char *s2, size_t n)
{
  char *s = s1;

  /* Find the end of S1.  */
  s1 += strlen (s1);

  size_t ss = __strnlen (s2, n);

  s1[ss] = '\0';
  memcpy (s1, s2, ss);

  return s;
}
```
```go
func strncat(s1 []byte, s2 []byte, n uint32) []byte {
        var s []byte = s1
        s1 += noarch.Strlen(s1)
        var// Warning (CallExpr): 34: probably an incorrect type translation 1 // <-- WRONG HERE
        ss uint32 = uint32((uint32(__strnlen(nil, n))))
        s1[ss] = '\x00'
        memcpy(s1, s2, uint32((ss)))
        return s
}
```

**Comment by Konstantin8105** [comment]
Closed, because it is part of #503
